### PR TITLE
Add subnetwork API with gateway IP

### PR DESCRIPTION
**Pull request by skmatti**
Also, add subnetworkName field to NetworkInterface API

### CODE REVIEW

File: apis/network/v1/networkinterface_types.go [hunk 1]
```diff
@@ -43,6 +43,10 @@ type NetworkInterfaceSpec struct {
 	// +kubebuilder:validation:MinLength=1
 	NetworkName string `json:"networkName"`
 
+	// SubnetworkName refers to a subnetwork of a network with in the cluster.
```
Comment (shouri007) [inline apis/network/v1/networkinterface_types.go:46]: `with in` -> `within`

File: apis/network/v1/subnetwork_types.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+// SubnetworkSpec is the desired configuration for the Subnetwork resource.
+type SubnetworkSpec struct {
+	// NetworkName refers to a network object that this Subnetwork is connected.
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	NetworkName string `json:"networkName"`
+	// Gateway4 defines the gateway IPv4 address for the Subnetwork.
+	// +optional
+	// +kubebuilder:validation:Format=ipv4
+	Gateway4 *string `json:"gateway4,omitempty"`
```
Comment (shouri007) [inline apis/network/v1/subnetwork_types.go:45]: would be good to explain the behavior when this field is not set

File: apis/network/v1/networkinterface_types.go (hunk 1 above)
Comment (skmatti) [reply apis/network/v1/networkinterface_types.go]: done

File: apis/network/v1/subnetwork_types.go (hunk 2 above)
Comment (skmatti) [reply apis/network/v1/subnetwork_types.go:48]: sure, added. WDYT?

**Pull request merged**